OpeningTimes.js

```diff
@@ -143,76 +152,27 @@ class OpeningTimes {
     return undefined;
   }
 
-  _findNextClosingTime(moment, sessions) {
-    const currentDay = sessions
-      .find((day) => (day.some(this._getDateInSessionFinder(moment))));
-
-    if (currentDay) {
-      const thisSession = currentDay.find(this._getDateInSessionFinder(moment));
-      return new Moment(thisSession.to).tz(this._timeZone);
-    }
-
-    return undefined;
-  }
-  /* Public API */
-
-  isOpen(moment) {
-    const sessions = this._getOpenSessions(moment);
-    return this._findMomentInSessions(moment, sessions);
-  }
-
-  nextOpen(moment) {
-    const allSessions = this._getOpenSessions(moment);
+  _nextOpen(moment) {
+    const allSessions = this._getOpenSessions(moment, [-1, 0, 1, 2, 3, 4, 5, 6]);
     if (this._findMomentInSessions(moment, allSessions)) {
       return moment;
     }
     return this._findNextOpeningTime(moment, allSessions);
   }
 
-  nextClosed(moment) {
-    const allSessions = this._getOpenSessions(moment);
-    if (!this._findMomentInSessions(moment, allSessions)) {
-      return moment;
-    }
 
-    return this._findNextClosingTime(moment, allSessions);
-  }
-
-  getOpeningHoursMessage(moment) {
-    if (this.isOpen(moment)) {
-      const closedNext = this.nextClosed(moment);
-      const closedTime = closedNext.format('h:mm a');
-      const closedDay = closedNext.calendar(moment, {
-        sameDay: '[today]',
-        nextDay: '[tomorrow]',
-        nextWeek: 'dddd',
-        lastDay: '[yesterday]',
-        lastWeek: '[last] dddd',
-        sameElse: 'DD/MM/YYYY',
-      });
-      return (
-        ((closedDay === 'tomorrow' && closedTime === '12:00 am')
-          || (closedDay === 'today' && closedTime === '11:59 pm')) ?
-        'Open until midnight' :
-        `Open until ${closedTime} ${closedDay}`);
-    }
-    const openNext = this.nextOpen(moment);
-    if (!openNext) {
-      return 'Call for opening times.';
+  /* Public API */
+
+  getStatus(moment, options = {}) {
+    const returnValue = { moment };
+
+    const session = this._getOpeningTimesSessionForMoment(moment, 1);
+    returnValue.isOpen = (session !== undefined);
+    if (options.until) {
+      returnValue.until = (session) ? session.to : this._nextOpen(moment);
```

Comment:
Any chance `until` could be renamed? Something along the lines of `nextStateChange` would be more descriptive.